driver-core/src/main/java/com/datastax/driver/core/AbstractAddressableByIndexData.java

```diff
@@ -256,22 +209,14 @@ public T setInet(int i, InetAddress v) {
     }
 
     public T setUDTValue(int i, UDTValue v) {
-        checkType(i, UDT);
         return setValue(i, codecFor(i, UDTValue.class).serialize(v, protocolVersion));
     }
 
     public T setTupleValue(int i, TupleValue v) {
-        checkType(i, TUPLE);
         return setValue(i, codecFor(i, TupleValue.class).serialize(v, protocolVersion));
     }
 
     @Override
-    public <V> T setObject(int i, V v) {
```

Comment:
> "Here's an object, I have no idea what this is, but let the driver figure out what to do with it"

I can't think of a use case where someone would want to throw random objects at the driver. And even so, they can still obtain the codec manually with `CodecRegistry#codecFor(DataType, T)` and then call `set(int, V, TypeCodec<V>)`.

The only reason we initially added setObject was to have a counterpart to getObject. But while getObject has a valid use case (stuff like a generic row logger), setObject does not.